packages/fetchai/skills/carpark_detection/handlers.py

```diff
@@ -244,12 +246,23 @@ def _handle_accept(self, msg: FIPAMessage, dialogue: Dialogue) -> None:
                 self.context.agent_name, msg.counterparty[-5:]
             )
         )
+        proposal = cast(Description, dialogue.proposal)
+        identifier = cast(str, proposal.values.get("ledger_id"))
+        self.context.ledger_apis.random_message = _generate_random_message(
+            nonce=new_message_id,
+            seller=self.context.agent_address,
+            client=msg.counterparty,
+            time_stamp=int(time.time()),
```

Comment:
note, you don't need time_stamp as an argument, you can just generate the time stamp in the helper method.